src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc

```diff
@@ -338,4 +295,169 @@ std::optional<edm4eic::MutableCluster> CalorimeterClusterRecoCoG::reconstruct(co
   return std::move(cl);
 }
 
+//------------------------------------------------------------------------
+void CalorimeterClusterRecoCoG::associate(
+  const edm4eic::Cluster& cl,
+#if EDM4EIC_VERSION_MAJOR >= 7
+  const edm4eic::MCRecoCalorimeterHitAssociationCollection* mchitassociations,
+#else
+  const edm4hep::SimCalorimeterHitCollection* mchits,
+#endif
+  edm4eic::MCRecoClusterParticleAssociationCollection* assocs
+) const {
+
+  // --------------------------------------------------------------------------
+  // Association Logic
+  // --------------------------------------------------------------------------
+  /*  1. identify all sim hits associated with a given protocluster, and sum
+   *     the energy of the sim hits.
+   *  2. for each sim hit, identify all contributing primary particles, and sum
```

Comment:
```suggestion
   *  2. for each sim hit, identify the parent of the contributing particle, and sum
```

Reply:
I reworked the description of the logic here: I tried to point out that we do look through the parents of each contributing particle for each sim hit, and that we only add it to the list of associations to make if the parent is a primary.

Let me know if there's a more clear way to phrase things...